src/RecyclableMemoryStream.cs

```diff
@@ -592,15 +596,16 @@ public override Task CopyToAsync(Stream destination, int bufferSize, Cancellatio
                     if (this.blocks.Count == 1)
                     {
                         AssertLengthIsSmall();
-                        return destination.WriteAsync(this.blocks[0], (int)this.position, (int)(this.length - this.position), cancellationToken);
+                        return destination.WriteAsync(this.blocks[0], (int)this.position, (int)count, cancellationToken).ContinueWith((task) => this.position += count);
```

Comment:
We can avoid creating a continuation by advancing the position first. (This is what MemoryStream does.)
```suggestion
                        var pos = this.position;
                        this.position += count;
                        return destination.WriteAsync(this.blocks[0], (int)pos, (int)count, cancellationToken);
```